backend/handleError.ts

```diff
@@ -31,7 +31,18 @@ export function handleError(res: NextApiResponse, e: unknown) {
   }
 
   if (e instanceof ZodError) {
-    return res.status(400).json(e);
+    if (e.format()._errors.length === 0) {
+      if (e.errors[0].code === 'invalid_type') {
+        const error = e.errors[0];
+        return res
+          .status(400)
+          .send(
+            `${error.path[0]} is ${error.message}. Received ${error.received} expected ${error.expected}. Zod error code: ${error.code}`,
+          );
+      }
+      return res.status(400).json(e.errors[0].message);
+    }
+    return res.status(400).json(e.format()._errors);
```

Comment:
Tämä toteutus liittyy siihen ideaan, että handlataan kaikki Zodit `.parse`a käyttäen eikä `.safeParse`a

Tämä kohta on hiukan vaikeahko selittää, mutta katsotaan mitä tulee :sweat_smile: 

Jos Zod huomasi virheen `z.object()`:in sisällä, `e.format()._errors`:in array oli tyhjä, joten fronttiin palautui tyhjä array. Teoriassa kyseistä virhettä ei pitäisi tapahtua sivua "normaalisti" käyttämällä, mutta parempi olla "safe than sorry"

Reply:
![kuva](https://github.com/user-attachments/assets/f172fa89-0197-4309-bff5-6331c3f7344b)

Halusin nopeasti näyttää kuvan avulla mitä meinasin :smile: 

Eli, lähetän kuvassa 1. requestin serverille (käsin), jossa haluan tehdä lahjalisäyksen.

Lahjan luonnin schema:
```ts
export const createGiftSchema = z.object({
  gift: z.string().min(1),
  receiver: z.string().min(1),
});
```
Kirjoitin tarkoituksella `receiver` -> **`receiverr`**, jotta tulee virhe

kuvassa 2. on vielä request selaimen Network-tabista Payloadin kohdasta napattu.

kuva 3. on kanssa napattu Network-tabista nyt Preview-kohdasta. Eli tuollaisena virhe tulisi ulos

kuva 4. on terminaalista napattu, eli tuolta virhe näyttää kun se tulee ja tuosta "kokosin" virhetekstin

Reply:
Ei hätää, itsekin lukasin aiemmin vain tuon otsikon "Formatting Errors" ja päättelin loput. Eli itsekin tein virheen :smile: 

Joo, tuo sun ratkaisu näyttäisi hyvältä! :+1: